### ISSUE TITLE
Respect single page reference in Indexer::___getPageReferenceIndexValue

### ISSUE THREAD

**Issue by Theominator** [teppokoivula/SearchEngine#10]
When indexing page references, only multiple pages (page arrays) are taken into account. In page reference field settings, one can choose the type "Single page", which is not considered here. With the follwing lines a single page will be converted into a page array for proper further processing (Indexer.php before line 217):

```
if ($page_ref instanceof \ProcessWire\Page) {
   $page_ref = \ProcessWire\PageArray([ $page_ref ]);
}
```

**Comment by mminglis** [comment]
Came here to say the same. Eg. a single PageReference used to select a category will be ignored.

**Comment by teppokoivula** [comment]
Thanks for the report!

This is fixed in SearchEngine 0.17.2. Looks like I assumed the unformatted value to be PageArray, probably since File fields work that way (i.e. unformatted value is always an array type). Silly mistake from my part 🙂

Just for the record, I ended up with a slightly different implementation for the fix: SearchEngine officially supports 3.0.112, while PageArray($page_ref) requires 3.0.123. Not a big difference and I may bump the version requirement soon anyway — but I also personally prefer the native "new object" syntax over (IMHO less obvious) function shortcuts 😅
